File: Python/test_sort_evaluator.py
#
#  Copyright (c) 2021 Kirk Nickish
#  All rights reserved.
#

from sortEvaluator import EvaluateSorter
import copy
import sys, os

# Disable
def blockPrint():
    sys.stdout = open(os.devnull, 'w')

# Restore
def enablePrint():
    sys.stdout = sys.__stdout__

def testSorter(list_arg):
    internal = list_arg.copy()
    internal.sort()
    return internal

def testFailSort(list_arg):
    return list_arg

blockPrint()
model = EvaluateSorter(testSorter)
model.evaluate(1000)


try:
    model.replace_sorter(testFailSort)
    model.evaluate(1000)
    raise Exception("Fail model passed evaluation?????")
except:
    enablePrint()

File: CPP/test_sort_evaluator.cpp
/*
 *   Copyright (c) 2021 Kirk Nickish
 *   All rights reserved.
 */

#include "sort_evaluator.hpp"

int cmpfunc (const void * a, const void * b) {
   return ( *(int*)a - *(int*)b );
}

vector<int>  known_good_sorter(int ele_count, vector<int> &int_array_arg)
{
    vector<int> int_array = int_array_arg;
    std::sort(int_array.begin(), int_array.end());
    return int_array;
}

int main(void)
{
    SortEvaluator ev;
    if(ev.evaluate(known_good_sorter))
        return 0;
    else;
        return 1;
}